### PR TITLE
First Connect SDK

### PR DESCRIPTION
**Pull request by masesdevelopers**
## Description
This PR introduces the SDK dedicated to the Apache Kafka Connect usable from .NET.

## Related Issue

#65 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/java/knet/src/main/java/org/mases/knet/connect/KNetConnectProxy.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class KNetConnectProxy {
+    private static final Logger log = LoggerFactory.getLogger(KNetSourceTask.class);
+
+    public static final String DOTNET_CLASSNAME_CONFIG = "classname";
```
Comment (mariomastrodicasa) [inline src/java/knet/src/main/java/org/mases/knet/connect/KNetConnectProxy.java:35]: Convert the property name from **classname** to **knet.classname**: in this way the specific properties of KNet will have a common prefix

File: src/java/knet/src/main/java/org/mases/knet/connect/sink/KNetSinkConnector.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    public void start(Map<String, String> props) {
+        try {
+            if (!KNetConnectProxy.initializeSinkConnector(props)) {
+                log.error("Failed Invoke of \"initializeSinkConnector\"");
+                throw new ConfigException("Failed Invoke of \"initializeSinkConnector\"");
+            } else {
+                JCOBridge.RegisterJVMGlobal("KNetSinkConnector", this);
+                try {
+                    dataToExchange = props;
+                    KNetConnectProxy.getSinkConnector().Invoke("StartInternal");
```
Comment (mariomastrodicasa) [inline src/java/knet/src/main/java/org/mases/knet/connect/sink/KNetSinkConnector.java:59]: Test return value of getSinkConnector()

File: src/java/knet/src/main/java/org/mases/knet/connect/source/KNetSourceConnector.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    public void start(Map<String, String> props) {
+        try {
+            if (!KNetConnectProxy.initializeSourceConnector(props)) {
+                log.error("Failed Invoke of \"initializeSourceConnector\"");
+                throw new ConfigException("Failed Invoke of \"initializeSourceConnector\"");
+            } else {
+                JCOBridge.RegisterJVMGlobal("KNetSourceConnector", this);
+                try {
+                    dataToExchange = props;
+                    KNetConnectProxy.getSourceConnector().Invoke("StartInternal");
```
Comment (mariomastrodicasa) [inline src/java/knet/src/main/java/org/mases/knet/connect/source/KNetSourceConnector.java:59]: Test result of getSourceConnector()

File: src/net/KNet/ClientSide/BridgedClasses/Connect/KNetConnectProxy.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+        public static void Register<TTask>(KNetSourceConnector<TTask> instance) where TTask : KNetSourceTask
+        {
+            lock (globalInstanceLock)
+            {
+                if (SourceConnector != null) throw new InvalidOperationException("Cannot register multiple KNetSourceConnector");
+                SourceConnector = instance;
+                KNetCore.GlobalInstance.RegisterCLRGlobal(SourceConnector.ConnectorName, SourceConnector);
+            }
+        }
```
Comment (mariomastrodicasa) [inline src/net/KNet/ClientSide/BridgedClasses/Connect/KNetConnectProxy.cs:97]: Are the functions used?

**Review by mariomastrodicasa — CHANGES_REQUESTED** [review]

File: src/net/KNet/ClientSide/BridgedClasses/Connect/KNetConnectProxy.cs (hunk 4 above)
Comment (masesdevelopers) [reply src/net/KNet/ClientSide/BridgedClasses/Connect/KNetConnectProxy.cs:97]: Seems they can be removed. The registration is done automatically when Java requests to allocate the connectors: this is the best way we found to accomplish a start-up in distributed mode.

**Review by mariomastrodicasa — APPROVED** [review]

**Pull request merged**